theseus/tests/test_theseus_layer.py

```diff
@@ -125,6 +125,7 @@ def error_fn(optim_vars, aux_vars):
                 th.Vector(cost_weight_dim, name="learnable_err_param", tensor=cw_data)
             ],
             autograd_vectorize=True,
+            autograd_mode="dense",
```

Comment:
@fantaosha In [this](https://github.com/facebookresearch/theseus/blob/main/theseus/core/cost_function.py#L243) line, `functorch` throws an error when the batch size is not the same for all variables. We could consider automatically broadcasting before passing to `vmap`. Roughly something like the code below (not a clean version)

<img width="647" alt="image" src="https://user-images.githubusercontent.com/4759586/193978602-44539103-8c58-41e6-8426-60ffd09394fd.png">

Any concerns about adding something along these lines in a separate PR?

Reply:
LGTM.